common/primitives/src/utils.rs

```diff
@@ -19,6 +19,32 @@ pub mod as_hex {
 	}
 }
 
+/// Handle serializing and deserializing from `Option<Vec<u8>>` to hexadecimal
+#[cfg(feature = "std")]
+pub mod as_hex_option {
+	use serde::{Deserializer, Serializer};
+
+	/// Serializes a `Vec<u8>` into a hexadecimal string
+	pub fn serialize<S>(bytes: &Option<Vec<u8>>, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: Serializer,
+	{
+		match bytes {
+			Some(bytes) => impl_serde::serialize::serialize(bytes.as_slice(), serializer),
+			None => serializer.serialize_none(),
+		}
+	}
+
+	/// Deserializes a hexadecimal string into a `Vec<u8>`
+	pub fn deserialize<'de, D: Deserializer<'de>>(
+		deserializer: D,
+	) -> Result<Option<Vec<u8>>, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		impl_serde::serialize::deserialize(deserializer).map(|r| Some(r))
```

Comment:
I think this is suppose to return Ok(Some(r))?

Reply:
`.map` takes care of wrapping the option type as a Result
<img width="1051" alt="Screen Shot 2022-09-30 at 3 30 57 PM" src="https://user-images.githubusercontent.com/40524916/193351744-a3e28b62-9014-4eda-bfbc-36af1225c4ce.png">